### PR TITLE
Facehugger Resprite

### PR DESCRIPTION
**Pull request by kugamo**
## About The Pull Request

This PR resprites the playable facehugger, now with creepy crawly animations! Also fixed a runtime when facehuggers tried scuttling under airlocks --> now allowing them to, but without the ability to pounce while doing so _(idk if thats a fix feature or qol but whatever)_.

## Why It's Good For The Game

Higher quality sprites for your viewing pleasure. 🙂 

image

https://user-images.githubusercontent.com/16618648/202915469-eddc586e-efc1-4a12-952e-50666db69379.mp4

## Changelog

:cl:
imageadd: New playable facehugger sprites, now with creepy crawly animations.
code: Facehuggers can now scuttle under doors, but cannot pounce while doing so; Fixing a runtime.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/abilities/facehugger/facehugger_powers.dm [hunk 1]
```diff
@@ -19,3 +19,10 @@
 	if(facehugger.pounce_distance <= 1 && can_hug(L, facehugger.hivenumber))
 		did_hug = facehugger.handle_hug(L)
 	log_attack("[key_name] [did_hug ? "successfully hugged" : "tried to hug"] [key_name(L)] (Pounce Distance: [facehugger.pounce_distance]) at [get_location_in_text(L)]")
+
+/datum/action/xeno_action/activable/pounce/facehugger/use_ability()
+	for(var/obj/structure/machinery/door/airlock/A in get_turf(owner))
+		if(A.density) //if its CLOSED YOU"RE SCUTTLING AND CANNOT POUNCE!!!
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/abilities/facehugger/facehugger_powers.dm:25]:
```suggestion
		if(A.density) //if its CLOSED YOU'RE SCUTTLING AND CANNOT POUNCE!!!
```

Comment (harryob) [reply code/modules/mob/living/carbon/xenomorph/abilities/facehugger/facehugger_powers.dm:25]: CM saved

File: code/modules/mob/living/carbon/xenomorph/castes/Facehugger.dm [hunk 2]
```diff
@@ -191,3 +195,19 @@
...
+			return
+	// Is it an airlock?
+	if(istype(S, /obj/structure/machinery/door/airlock))
+		var/obj/structure/machinery/door/airlock/A = S
+		if(A.locked || A.welded) //Can't pass through airlocks that have been bolted down or welded
+			to_chat(src, SPAN_WARNING("\The [A] is locked down tight. You can't squeeze underneath!"))
+			return
+	visible_message(SPAN_WARNING("\The [src] scuttles underneath \the [S]!"), \
+	SPAN_WARNING("You squeeze and scuttle underneath \the [S]."), null, 5)
+	forceMove(S.loc)
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/castes/Facehugger.dm:213]:
```suggestion
/mob/living/carbon/Xenomorph/Facehugger/proc/scuttle(var/obj/structure/current_structure)
	var/move_dir = get_dir(src, loc)
	for(var/atom/movable/atom in get_turf(current_structure))
		if(atom != current_structure && atom.density && atom.BlockedPassDirs(src, move_dir))
			to_chat(src, SPAN_WARNING("\The [atom] prevents you from squeezing under \the [current_structure]!"))
			return
	// Is it an airlock?
	if(istype(current_structure, /obj/structure/machinery/door/airlock))
		var/obj/structure/machinery/door/airlock/current_airlock = current_structure
		if(current_airlock.locked || current_airlock.welded) //Can't pass through airlocks that have been bolted down or welded
			to_chat(src, SPAN_WARNING("\The [airlock] is locked down tight. You can't squeeze underneath!"))
			return
	visible_message(SPAN_WARNING("\The [src] scuttles underneath \the [current_structure]!"), \
	SPAN_WARNING("You squeeze and scuttle underneath \the [current_structure]."), null, 5)
	forceMove(current_structure.loc)
```

i love descriptive var names

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/abilities/facehugger/facehugger_powers.dm (hunk 1 above)
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/abilities/facehugger/facehugger_powers.dm:25]:
```suggestion
	for(var/obj/structure/machinery/door/airlock/current_airlock in get_turf(owner))
		if(current_airlock.density) //if its CLOSED YOU'RE SCUTTLING AND CANNOT POUNCE!!!
```
i forgor

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by monkeysfist101 — APPROVED** [review]
Disgusting. Approved.

**Comment by LoJoFan** [comment]
https://user-images.githubusercontent.com/112346996/203282026-6a7f5181-2edf-4751-9b49-b1cf50907c8a.mp4

**Pull request merged**